compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -1125,19 +1119,26 @@ pub fn describe_flag_categories(early_dcx: &EarlyDiagCtxt, matches: &Matches) ->
     }
 
     if cg_flags.iter().any(|x| *x == "passes=list") {
-        let backend_name = debug_flags.iter().find_map(|x| x.strip_prefix("codegen-backend="));
-
-        let opts = config::Options::default();
-        let sysroot = filesearch::materialize_sysroot(opts.maybe_sysroot.clone());
-        let target = config::build_target_config(early_dcx, &opts, &sysroot);
-
-        get_codegen_backend(early_dcx, &sysroot, backend_name, &target).print_passes();
+        get_backend_from_raw_matches(early_dcx, matches).print_passes();
         return true;
     }
 
     false
 }
 
+fn get_backend_from_raw_matches(
```

Comment:
let's add a doc comment explaining why this exists

Reply:
Added

Reply:
Looks good, thank you.

Reply:
Err, did you mean "creates a sessions" or "created a session", I assume the latter.